src/gurt/misc.c

```diff
@@ -11,12 +11,41 @@
 
 #include <stdarg.h>
 #include <math.h>
+#include <malloc.h>
 #include <string.h>
 #include <gurt/common.h>
 
 /* state buffer for DAOS rand and srand calls, NOT thread safe */
 static struct drand48_data randBuffer = {0};
 
+d_alloc_track_cb_t d_alloc_track_cb;
+d_alloc_track_cb_t d_free_track_cb;
+static __thread void *track_arg;
+
+#define D_MEMORY_TRACK_ENV "D_MEMORY_TRACK"
+
+void
+d_set_alloc_track_cb(d_alloc_track_cb_t alloc_cb, d_alloc_track_cb_t free_cb)
+{
+	bool	track_mem = false;
+
+	d_getenv_bool(D_MEMORY_TRACK_ENV, &track_mem);
+	if (unlikely(track_mem)) {
+		d_alloc_track_cb = alloc_cb;
+		d_free_track_cb = free_cb;
+		D_INFO("memory track is enabled for the engine.\n");
+	}
+}
+
+void
+d_set_alloc_track_arg(void *arg)
```

Comment:
Is there any reason to export two separated APIs for callback & callback arg registering? Why don't we just register both callback & callback arg by single register API?

Making caller to separately register a callback and a per-thread callback arg looks weird to me. As for this PR, I think we don't need to register callback arg, since we could get the per-thread 'mem_stats' in dss_mem_total_alloc_track(), right?